### DISCUSSION TITLE
Move data about sidebars and headers from HTML to YAML

### DISCUSSION THREAD

**Discussion by bershanskiy** [mdn/yari#2459] [Ideas]
MDN has a lot of macros which insert headers and sidebars, including:
 - `Draft`
 - `SeeCompatTable`
 - sidebars like `AddonSidebar`, `ToolsSidebar`, `APIRef`, etc.
 - `securecontext_header`
 - `AvailableInWorkers`

I think it's possible to move these declarations to YAML header. Then later we could even cross-check some of them with BCD.

## Examples

| Macro               | YAML equivalent | Notes             |
|------------------|--------------------|----------------|
| `Draft`           | `draft: true`          | Like in Hugo frontmatter, if `draft` is omitted or `false` document is not draft. Any value besides `true` and `false` is invalid. |
| `non-standard_header` | `standard_track: false` | Like BCD's `standard_track` |
| `deprecated_header` | `deprecated: true` | Like BCD's `deprecated` |
| `SeeCompatTable` | `experimental: true` | Like BCD's `experimental` |
| `AddonSidebar` | `sidebar: addon` | |
| `ToolsSidebar` | `sidebar: tool` |
| `APIRef("_string_")` | `sidebar: "_string_"` | Complete list of permitted strings inferred from content. |

**Comment by peterbe** [comment]
We could start with just setting the sidebar in the front-matter. Other KS macros like `non-standard_header` feels more OK to leave to another day. The reason is that the sidebars are treated very differently in Yari, but things like `{{SeeCompatTable}}` are just HTML as far as Yari is concerned. 

What's exciting to me is the potential to rewrite the sidebars. Instead of generating the sidebars from a KS macro (with full HTML) we can instead write a new piece of Node code that traverses the tree of files and extracts all the URLs + titles and then groups them into a plain array.
